src/plugins/embeddable/public/lib/panel/embeddable_panel_error.tsx

```diff
@@ -62,42 +59,29 @@ export function EmbeddablePanelError({
 
     return () => subscription.unsubscribe();
   }, [editPanelAction, embeddable]);
-  useEffect(() => {
-    if (!ref.current) {
-      return;
-    }
-
-    if (!embeddable.catchError) {
-      const errorEmbeddable = new ErrorEmbeddable(error, { id: embeddable.id });
-      setNode(errorEmbeddable.render());
-
-      return () => errorEmbeddable.destroy();
-    }
-
-    const renderedNode = embeddable.catchError(error, ref.current);
-    if (isFunction(renderedNode)) {
-      return renderedNode;
-    }
-    if (isPromise(renderedNode)) {
-      renderedNode.then(setNode);
-    } else {
-      setNode(renderedNode);
-    }
-  }, [embeddable, error]);
 
   return (
-    <EuiPanel
-      element="div"
-      className="embPanel__content embPanel__content--error"
-      color="transparent"
-      paddingSize="none"
-      data-test-subj="embeddableError"
-      panelRef={ref}
-      role={isEditable ? 'button' : undefined}
-      aria-label={isEditable ? ariaLabel : undefined}
-      onClick={handleErrorClick}
-    >
-      {node}
-    </EuiPanel>
+    <EuiEmptyPrompt
+      body={
+        <EuiText size="xs">
+          <Markdown
+            markdown={error.message}
+            openLinksInNewTab={true}
+            data-test-subj="errorMessageMarkdown"
+          />
+        </EuiText>
+      }
+      data-test-subj="embeddableStackError"
+      iconType="alert"
+      iconColor="danger"
+      layout="vertical"
+      actions={
+        isEditable && (
+          <EuiButton color="primary" fill onClick={handleErrorClick} size="s">
+            {label}
```

Comment:
In the screenshot you shared, I noticed that the button text could be `Edit lens` or `Edit Visualization`. Saying `Edit lens` sounds a little strange (since Lens is the name of the app, not the thing being edited). The word "Lens" should also be uppercase. Similarly, the word "visualization" should be lowercase. Is it possible to change that?

Alternatively, if it is always visualizations that are being edited, we could make the button text a static `Edit visualization` for consistency. Thoughts?

Reply:
Adding on to what @MichaelMarcialis said, this button changes to `Edit <title>` when the embeddable is by reference:

![image](https://user-images.githubusercontent.com/8698078/197859459-3f671320-99d4-4513-9f06-32a673dadead.png)

Definitely think we should try to unify this so that all visualizations behave the same, regardless of if they are saved to the library or what editor was used.